lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
```

Comment:
It definitely seems weird to use `prefetch` for this purpose, I agree... 

I think that the most correct thing is probably to use a feature, which I had kind of forgotten about.  I moved the code over to a feature and made a few tweaks to make it run, but haven't modified tests yet.  I pushed it up to the branch in my repo (note I haven't fixed tests yet):

https://github.com/ferventcoder/puppetlabs-dsc/compare/ticket/master/MODULES-2175_symlinks...Iristyle:pr/49?expand=1

I think a feature is probably the right place for this.  Then we also address the @cyberious point about separating the "vendoring through symlink" code from the provider.  And I believe the feature initialization only runs once as well.